source/synthDrivers/sapi5.py

```diff
@@ -31,16 +31,29 @@
 
 class FunctionHooker(object):
 
-	def __init__(self,targetDll,importDll,funcName,newFunction):
-		hook=NVDAHelper.localLib.dllImportTableHooks_hookSingle(targetDll,importDll,funcName,newFunction)
-		if hook:
-			log.debug("hooked %s"%funcName)
+	def __init__(
+		self,
+		targetDll: str,
+		importDll: str,
+		funcName: str,
+		newFunction # result of ctypes.WINFUNCTYPE
+	):
+		# dllImportTableHooks_hookSingle expects byte strings.
+		self._hook=NVDAHelper.localLib.dllImportTableHooks_hookSingle(

```

Comment:
In `nvdaHelper/local/dllImportTableHooks.cpp` we are using ANSI functions, which should probably change in the future, but for now we need to ensure that these strings are no more than ANSI strings. 

So I would use `encoding=ANSI` for each of these encode calls. We want to know about it if a we path can not be decoded by one of the windows functions. Probably want to wrap the calls to encode in a try block too.

Reply:
I chose for mbcs, which is what we use in other places.

Reply:
Yep, makes sense.